File: DIC/WebServices/kubeless/java-code/hash/Hash.java
package io.kubeless;

import io.kubeless.Event;
import io.kubeless.Context;
import com.google.gson.JsonObject;
import java.util.Calendar;

public class Hash {
    public String foo(io.kubeless.Event event, io.kubeless.Context context) {
        long startTime = Calendar.getInstance().getTimeInMillis();
        
        String hash1 = "hashtest";
        int token = hash1.hashCode();

        JsonObject response= new JsonObject();
        response.addProperty("token", token);
        response.addProperty("startTime",startTime);
        return response.toString();
    }
}

File: DIC/WebServices/kubeless/nodejs-code/hash/hash.js

String.prototype.hashCode = function(){
    if (Array.prototype.reduce){
        return this.split("").reduce(function(a,b){a=((a<<5)-a)+b.charCodeAt(0);return a&a},0);
    }
    var hash = 0;
    if (this.length === 0) return hash;
    for (var i = 0; i < this.length; i++) {
        var character  = this.charCodeAt(i);
        hash  = ((hash<<5)-hash)+character;
        hash = hash & hash;
    }
    return hash;
}

module.exports = {
  foo: function (event, context) {
      startTime = Date.now();
      var token = new String(event.data).hashCode();
      return {token: token, startTime: startTime}
    }
}